src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -165,6 +166,37 @@ public unsafe BitArray(bool[] values)
                     }
                 }
             }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                Vector128<byte> elementMask = Vector128.Create(0x01020408_10204080).AsByte();
```

Comment:
Are there endianness concerns here?

Reply:
Yes, I believe so as ARM doesn't have guaranteed endianness?

Given this `byte`, we should be able to use the `ReadOnlySpan<byte>` trick and a `Load` instead...

Reply:
@GrabYourPitchforks Do we run on anything other than little endian platforms? (If so, which ones?) Do we aspire to?

Reply:
Hmmmmmmm, I think that is a valid concern.

> Given this byte, we should be able to use the ReadOnlySpan<byte> trick and a Load instead...

In the last PR where I vectorised BitArray for x86, I think we decided that it is better to make it a static readonly field and just use that instead of doing this whole `ReadOnlySpan<byte>` thing and load: https://github.com/dotnet/corefx/pull/41896#discussion_r337583744

 >>  The most natural way to do this is using static readonly Vector256<byte>. I understand that the JIT emits a few extra instructions for this today. We should fix that if it matters.

Reply:
@BruceForstall @GrabYourPitchforks correct me if I am wrong but the only BE target at the moment - is classic (non dotnet/runtime one) mono which uses some of the dotnet/corefx sources (but 2.1)

Reply:
Then we should at minimum have a `Debug.Assert` every place where we make platform assumptions like this. Otherwise if we do want to target big endian platforms in the future we'll end up spending considerable time tracking down random failures.

Reply:
I certainly can look into making it better. It won't hurt much to consider endianness if there's a possibility... or if it is we can have an assertion at minimum.

Reply:
> Then we should at minimum have a Debug.Assert every place where we make platform assumptions like this. Otherwise if we do want to target big endian platforms in the future we'll end up spending considerable time tracking down random failures.

I agree, it is relatively trivial to support big endian here and unblocks any future work or platforms we want to support.

> In the last PR where I vectorised BitArray for x86, I think we decided that it is better to make it a static readonly field and just use that instead of doing this whole ReadOnlySpan<byte> thing and load: dotnet/corefx#41896 (comment)

Yeah, we should continue with that approach for this.

Reply:
Should be addressed now.